cmd/aws-vpc-cni-init/main.go

```diff
@@ -0,0 +1,168 @@
+	"github.com/aws/amazon-vpc-cni-k8s/utils/imds"
+	"github.com/aws/amazon-vpc-cni-k8s/utils/sysctl"
+	log "github.com/sirupsen/logrus"
+	"github.com/vishvananda/netlink"
+)
+
+const (
+	defaultHostCNIBinPath = "/host/opt/cni/bin"
+	metadataLocalIP       = "local-ipv4"
+	metadataMAC           = "mac"
+
+	envDisableIPv4TcpEarlyDemux = "DISABLE_TCP_EARLY_DEMUX"
+	envEnableIPv6               = "ENABLE_IPv6"
+	envHostCniBinPath           = "HOST_CNI_BIN_PATH"
+)
+
+func getEnv(env, def string) string {
+	if val, ok := os.LookupEnv(env); ok {
+		return val
+	}
+	return def
+}
+
+func main() {
+	os.Exit(_main())
+}
+
+func _main() int {
+	log.Debug("Started Initialization")
+	pluginBins := []string{"loopback", "portmap", "bandwidth", "aws-cni-support.sh"}
+	var err error
+	for _, plugin := range pluginBins {
+		if _, err = os.Stat(plugin); err != nil {
+			log.WithError(err).Fatalf("Required executable : %s not found\n", plugin)
+			return 1
+		}
+	}
+
+	hostCNIBinPath := getEnv(envHostCniBinPath, defaultHostCNIBinPath)
+
+	log.Infof("Copying CNI plugin binaries ...")
+	err = cp.InstallBinaries(pluginBins, hostCNIBinPath)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to install binaries")
+		return 1
+	}
+
+	log.Infof("Copied all CNI plugin binaries to %s\n", hostCNIBinPath)
+
+	var hostIP string
+	hostIP, err = imds.GetMetaData("local-ipv4")
+	if err != nil {
+		log.WithError(err).Fatalf("aws-vpc-cni init failed\n")
+		return 1
+	}
+
+	var primaryMAC string
+	primaryMAC, err = imds.GetMetaData("mac")
+	if err != nil {
+		log.WithError(err).Fatalf("aws-vpc-cni init failed\n")
+		return 1
+	}
+
+	log.Infof("Found hostIP %s and primaryMAC %s", hostIP, primaryMAC)
+
+	links, err := netlink.LinkList()
+	if err != nil {
+		log.WithError(err).Fatalf("Failed to list links\n")
+		return 1
+	}
+
+	var primaryIF string
+	for _, link := range links {
+		if link.Attrs().HardwareAddr.String() == primaryMAC {
+			primaryIF = link.Attrs().Name
+			break
+		}
+	}
+
+	if primaryIF == "" {
+		log.Errorf("Failed to retrieve primary IF")
+		return 1
+	}
+
+	log.Infof("Found primaryIF %s", primaryIF)
+	sys := sysctl.New()
+	entry := "net/ipv4/conf/" + primaryIF + "/rp_filter"
+	err = sys.SetSysctl(entry, 2)
+	if err != nil {
+		log.WithError(err).Fatalf("Failed to set rp_filter for %s\n", primaryIF)
+		return 1
+	}
+
+	val, _ := sys.GetSysctl(entry)
+	log.Infof("Updated entry for %d", val)
+
+	disableIPv4EarlyDemux := getEnv(envDisableIPv4TcpEarlyDemux, "false")
+	entry = "net/ipv4/tcp_early_demux"
+	if disableIPv4EarlyDemux == "true" {
+		err = sys.SetSysctl(entry, 0)
+		if err != nil {
+			log.WithError(err).Fatalf("Failed to disable tcp_early_demux\n")
+			return 1
+		}
+	} else {
+		err = sys.SetSysctl(entry, 1)
+		if err != nil {
+			log.WithError(err).Fatalf("Failed to enable tcp_early_demux\n")
+			return 1
+		}
+	}
+
+	val, _ = sys.GetSysctl(entry)
+	log.Infof("Updated entry for %d", val)
+
+	enableIPv6 := getEnv(envEnableIPv6, "false")
```

Comment:
nit: I would use a boolean here and cast the env var value if it is a string

Reply:
I am reusing getEnv which internally returns a string retrieved from the env variables hence I retained it.. The getEnv is used in multiple places since we would need a string say for instance to update conflist file.